app/views/assigned/clients/_client_table.haml

```diff
@@ -1,24 +1,25 @@
 .o-page__main-content.mt-3
   .clients__list
-    - clients.each do |user_client|
-      - client = user_client.client
+    - source_clients = clients.map(&:client)
+    - current_user.client_view_accessor.preload_source_clients(source_clients)
```

Comment:
maybe `client_view_accessor` should take an array of destination clients as an optional initialization argument?  Then this could be something like:

```
- view_clients = current_user.client_view_accessor(clients: clients)
...
- akas = view_clients.searchable_client_names(client).to_a
```

Essentially allowing the preload to be private.

Reply:
While it would be more concise, this API feels less clear to me. Exposing 'preload' feels clear and flexible. Packing the clients array into the accessor for the adapter feels ambiguous to me and raises questions around how I'm supposed to interact with the cache. Also with this api you'd need to pass in the whole client array to use the name accessor, which feels like extra work.

I think I can do a better job communicating the function of this class, I'll try and enrich the inline docs. Or happy to huddle if you have some suggestions

Reply:
per conversation, renamed preload_source_clients to preload_searchable_clients